Create “Join Newsletter!” Desktop

rapterjet2004 opened the pull request:
fixes #109 

- Design is consistent with Figma design
- Header is H1 “Join Our Newsletter!”
- Front is secondary light
- Background is Primary
- Shadow is Primary Dark
- Has fields: Email and Submit aligned next to each other
- added to social media page
- also fixed a small layout bug in DesktopSmBubbles

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,14 @@
+        <div className="relative h-4/6 w-1/2 bg-primary-dark rounded-[2.813rem] right-20 bottom-8"/>
+        <div className="absolute  h-4/6 w-1/2 pl-7 bg-secondary-light rounded-[2.813rem]"> 
+            <h1 className="mt-14"> Join Our Newsletter!</h1>
+            <p className="w-5/6 "> Want to get updates as fast as possible! Join our newsletter! We will send details about the event with sign up and RSVP forms as we get closer to the start date! We will also send insights into what types of events we are planning, and much more! Sign up to make sure you get the latest as soon as possible! </p>
```
Stylistic thing and I know it was not in the Figma design, but could we lighten up the paragraph text a bit to make it less contrasty on the eyes?

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,14 @@
+export default function DesktopNewsletter() {
+    return(
+    <div className="grid h-screen bg-primary place-items-center">
+        <div className="relative h-4/6 w-1/2 bg-primary-dark rounded-[2.813rem] right-20 bottom-8"/>
+        <div className="absolute  h-4/6 w-1/2 pl-7 bg-secondary-light rounded-[2.813rem]"> 
```
For some reason in my browser (safari) the top box is stuck on the left side of the screen. If it works for you, then try downloading another browser such as Firefox or Chrome and see if it still works.

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,14 @@
+        <div className="absolute  h-4/6 w-1/2 pl-7 bg-secondary-light rounded-[2.813rem]"> 
+            <h1 className="mt-14"> Join Our Newsletter!</h1>
+            <p className="w-5/6 "> Want to get updates as fast as possible! Join our newsletter! We will send details about the event with sign up and RSVP forms as we get closer to the start date! We will also send insights into what types of events we are planning, and much more! Sign up to make sure you get the latest as soon as possible! </p>
+            <input type="email" placeholder="Email:" className="mt-7 w-2/3 py-2 px-4 rounded-[2.813rem]"/>
```
This needs to be extracted into its own component per #112 . If you would like to help him out on this, since from a design standpoint this is pretty close. The only thing that I would like to see is a stronger border color.

ChristianBingman reviewed (changes requested):
Pretty close to what we are looking for, but the main things are some simple stylistic changes, and extracting out the input field.

rapterjet2004 commented:
Ok, I added lightened up the text to `#757575` and redid the layout. Not sure why it wasn't working on safari since I'm on windows, but I tested it on firefox, brave, and edge and it seems to be fine on my end.

 I'm just waiting on @kaodik  to finish his input text box to replace mine with. In the meanwhile I updated the border to reflect the figma design and made the button change to primary-dark on hover.

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,16 @@
+    return(
+    <div className="grid h-screen bg-primary place-items-center">
+        <div className="relative h-full w-full items-center justify-center inline-flex">
+            <div className="relative h-4/6 w-1/2 bg-primary-dark rounded-[2.813rem] bottom-10 right-10"/>
+            <div className="absolute h-4/6 w-1/2 pl-7 bg-secondary-light rounded-[2.813rem]"> 
```
Instead of doing this can we use h-fit and drop shadow to get the same effect? This way on smaller screens that are still desktops things don't get shoved outside the box, and when the screen is large, the box doesn't have a ton of space around it?
filter: drop-shadow(-90px -60px 4px rgba(0, 0, 0, 0.5));

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,16 @@
+            <div className="absolute h-4/6 w-1/2 pl-7 bg-secondary-light rounded-[2.813rem]"> 
+                <h1 className="mt-14"> Join Our Newsletter!</h1>
+                <p className="text-[#757575] w-5/6"> Want to get updates as fast as possible! Join our newsletter! We will send details about the event with sign up and RSVP forms as we get closer to the start date! We will also send insights into what types of events we are planning, and much more! Sign up to make sure you get the latest as soon as possible! </p>
+                <input type="email" placeholder="Email:" className="mt-7 w-2/3 py-2 px-4 border-2 text-[#757575] border-[#AAAAAA] rounded-[2.813rem]"/>
```
You should be able to use the input box styling after you rebase or merge

ChristianBingman reviewed (changes requested):
Pretty close just an idea on how to make it better.

rapterjet2004 commented:
Ok I made the div have a drop down shadow and I changed the input textbox to use Steven's implementation. Ignore the messy git history I screwed up the rebase but I fixed it now so everything should be fine.

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,18 @@
+            <div className="absolute h-fit w-1/2 pl-7 py-7 drop-shadow-[-90px_-60px_4px_rgba(0,0,0,0.5)] bg-secondary-light rounded-[2.813rem]"> 
+                <h1 className="mt-14"> Join Our Newsletter!</h1>
+                <p className="text-[#757575] w-5/6 py-2"> Want to get updates as fast as possible! Join our newsletter! We will send details about the event with sign up and RSVP forms as we get closer to the start date! We will also send insights into what types of events we are planning, and much more! Sign up to make sure you get the latest as soon as possible! </p>
+                <div className="grid grid-flow-col">                    
+                    <input type="text" placeholder="Email:" />
+                    <button className="ml-3 bg-primary-light hover:bg-primary-dark px-8 w-fit text-white font-bold drop-shadow-lg rounded-[2.813rem]" > Submit </button>
```
Could you make this fill the width of the box

ChristianBingman replied:
We already have a button component called PrimaryButton. You should use this instead of your own custom styling.

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,18 @@
+            {/* <div className="relative h-4/6 w-1/2 bg-primary-dark rounded-[2.813rem] bottom-10 right-10"/> */}
+            <div className="absolute h-fit w-1/2 pl-7 py-7 drop-shadow-[-90px_-60px_4px_rgba(0,0,0,0.5)] bg-secondary-light rounded-[2.813rem]"> 
+                <h1 className="mt-14"> Join Our Newsletter!</h1>
+                <p className="text-[#757575] w-5/6 py-2"> Want to get updates as fast as possible! Join our newsletter! We will send details about the event with sign up and RSVP forms as we get closer to the start date! We will also send insights into what types of events we are planning, and much more! Sign up to make sure you get the latest as soon as possible! </p>
```
Remove the width restriction

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,18 @@
+        <div className="relative h-full w-full items-center justify-center inline-flex">
+            {/* <div className="relative h-4/6 w-1/2 bg-primary-dark rounded-[2.813rem] bottom-10 right-10"/> */}
+            <div className="absolute h-fit w-1/2 pl-7 py-7 drop-shadow-[-90px_-60px_4px_rgba(0,0,0,0.5)] bg-secondary-light rounded-[2.813rem]"> 
+                <h1 className="mt-14"> Join Our Newsletter!</h1>
```
Remove the top margin

ChristianBingman commented on src/components/DesktopNewsletter/index.jsx:
```diff
@@ -0,0 +1,18 @@
+    <div className="grid h-screen bg-primary place-items-center">
+        <div className="relative h-full w-full items-center justify-center inline-flex">
+            {/* <div className="relative h-4/6 w-1/2 bg-primary-dark rounded-[2.813rem] bottom-10 right-10"/> */}
+            <div className="absolute h-fit w-1/2 pl-7 py-7 drop-shadow-[-90px_-60px_4px_rgba(0,0,0,0.5)] bg-secondary-light rounded-[2.813rem]"> 
```
Set pl-7 to px-7, and py-7 to py-10

ChristianBingman reviewed (approved):
Now just some minor style changes.

ChristianBingman reviewed (approved):
Looks good to me, thanks!
